plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/model/LibraryFactory.java

```diff
@@ -107,6 +107,11 @@ static Library create(IConfigurationElement configurationElement) throws Library
         MavenCoordinates mavenCoordinates = getMavenCoordinates(
             libraryFileElement.getChildren(ELEMENT_NAME_MAVEN_COORDINATES));
         LibraryFile libraryFile = loadSingleFile(libraryFileElement, mavenCoordinates);
+        // Pin servlets, JSP, and guava; update everything else
+        if (!libraryFile.getMavenCoordinates().getGroupId().startsWith("javax.servlet")
```

Comment:
I'd prefer modifying the plugin.xml definitions to support expressing pinning: there's no traceability from the library definitions to this code.  Especially since there are multiple javax.servlet* groups (e.g., javax.servlet, javax.servlet.jsp, javax.servlet.jsp.jstl), and who knows which group we'll need to select in the future.

Reply:
Good idea. Done.